client/controller/index.web.js

```diff
@@ -147,7 +147,10 @@ export const redirectInvalidLanguage = ( context, next ) => {
 
 export function redirectLoggedOut( context, next ) {
 	const state = context.store.getState();
-	if ( isUserLoggedIn( state ) ) {
+	const isLoggedIn = isUserLoggedIn( state );
+
+	// If the user is logged in - or logged out but coming from My Jetpack - continue
+	if ( isLoggedIn || ( ! isLoggedIn && context.query?.source === 'my-jetpack' ) ) {
```

Comment:
We rely on those middleware functions (e.g. `redirectLoggedOut`, `redirectLoggedOutToSignup`, etc) to delegate certain responsibilities to a single place. What I could do is create a separate middleware (`redirectLoggedOutIfNotFromMyJetpack`? 😅) and have only that condition, in case it's not from My Jetpack, it calls `redirectLoggedOut`. I initially preferred to just extend the condition, but I'm open to change it to something else. What do you think?
About being specific: there are other code snippets that handle specific cases there, so I don't think it's a big problem.

Reply:
Ok yeah I think it's better than having a separate middleware for one case 😅 It seems fine to me with that explanation 😄

Reply:
I left some feedback in a separate review, but I think a separate middleware function might be a nice solution - I think we'd have fewer files to edit and it would be easier to follow the logic contained in that one place.